platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/utility/RecycleBin.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.io.utility;
+
+import static com.swirlds.logging.LogMarker.EXCEPTION;
+
+import com.swirlds.common.config.StateConfig;
+import com.swirlds.common.io.config.RecycleBinConfig;
+import com.swirlds.common.system.NodeId;
+import com.swirlds.common.threading.locks.AutoClosableLock;
+import com.swirlds.common.threading.locks.internal.AutoLock;
+import com.swirlds.common.threading.locks.locked.Locked;
+import com.swirlds.config.api.Configuration;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Objects;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * This class provides the abstraction of deleting a file, but actually moves the file to a temporary location in case
+ * the file becomes useful later for debugging.
+ */
+public class RecycleBin {
+
+    private static final Logger logger = LogManager.getLogger(RecycleBin.class);
+
+    private final Path recycleBinPath;
+    private final AutoClosableLock lock = new AutoLock();
```

Comment:
Please use Locks as factory class. By doing so you you do not need to depend on specific implementations.

Reply:
done